solana-programs/audius_eth_registry/src/processor.rs

```diff
@@ -132,6 +266,278 @@ impl Processor {
             .map_err(|e| e.into())
     }
 
+    /// Process [ValidateMultipleSignaturesClearValidSigner]().
+    pub fn process_multiple_signatures_clear_valid_signer(
+        accounts: &[AccountInfo],
+        signature_data_1: SignatureData,
+        signature_data_2: SignatureData,
+        signature_data_3: SignatureData,
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        // initialized valid signer account 1
+        let valid_signer_1_info = next_account_info(account_info_iter)?;
+        // initialized valid signer account 2
+        let valid_signer_2_info = next_account_info(account_info_iter)?;
+        // initialized valid signer account 3
+        let valid_signer_3_info = next_account_info(account_info_iter)?;
+        // signer group account
+        let signer_group_info = next_account_info(account_info_iter)?;
+        // incoming valid signer account
+        let old_valid_signer_info = next_account_info(account_info_iter)?;
+
+        // Confirm both signatures are valid
+        // Sysvar Instruction account info
+        let instruction_info = next_account_info(account_info_iter)?;
+
+        // clock sysvar account
+        let clock_account_info = next_account_info(account_info_iter)?;
+        let clock = Clock::from_account_info(&clock_account_info)?;
+
+        // Index of current instruction in tx
+        let index = sysvar::instructions::load_current_index(&instruction_info.data.borrow());
+
+        if index == 0 {
+            return Err(AudiusError::Secp256InstructionLosing.into());
+        }
+
+        // Instruction data of 1st Secp256 program call
+        let secp_instruction_1 = sysvar::instructions::load_instruction_at(
```

Comment:
Probably worth refactoring to do this in a cycle.

Reply:
wdym by a cycle here?